module/zcommon/zfs_fletcher.c

```diff
@@ -300,33 +300,47 @@ fletcher_2_byteswap(const void *buf, uint64_t size,
 	(void) fletcher_2_incremental_byteswap((void *) buf, size, zcp);
 }
 
-ZFS_NO_SANITIZE_UNDEFINED
+#if defined(__GNUC__) && !defined(__clang__)
+__attribute__((optimize("no-tree-vectorize")))
+#elif defined(__clang__)
+__attribute__((optnone))
+#endif
```

Comment:
I see the zstd code defines a `DONT_VECTORIZE` macro in `module/zstd/lib/common/compiler.h`.  We could define our own version of this for use throughout the code.

As you already mentioned I think we should update the superscaler implementations as well.